python_ta/transforms/type_inference_visitor.py

```diff
@@ -146,63 +146,49 @@ def visit_name(self, node):
     ##############################################################################
     # Operation nodes
     ##############################################################################
-    def visit_binop(self, node):
-        t1 = self.type_constraints.lookup_concrete(node.left.type_constraints.type)
-        t2 = self.type_constraints.lookup_concrete(node.right.type_constraints.type)
-        op_name = op_to_dunder_binary(node.op)
-
+    def _lookify_call(self, node, func_name, *args):
+        """Helper to lookup a function and unify it with given arguments.
+           Returns the return type of unified function call."""
+        arg_types = [self.type_constraints.lookup_concrete(arg) for arg in args]
         try:
-            method_type = self.type_store.lookup_function(op_name, t1, t2)
+            func_type = self.type_store.lookup_function(func_name, *arg_types)
         except KeyError:
-            node.type_constraints = TypeInfo(
-                TypeErrorInfo('Method {}.{}({}) not found'.format(t1, op_name, t2), node)
-            )
-            return
+            return_info = TypeInfo(
+                TypeErrorInfo('Function {} not found with given a {}'.
+                              format(func_name, *arg_types), node))
+            return return_info
 
         try:
-            return_type = self.type_constraints.unify_call(method_type, t1, t2)
+            return_type = self.type_constraints.unify_call(func_type, *arg_types)
         except TypeInferenceError:
-            node.type_constraints = TypeInfo(
-                TypeErrorInfo('incompatible types {} and {} in BinOp'.format(t1, t2), node)
-            )
+            return_info = TypeInfo(
+                TypeErrorInfo('Bad unify_call of Function {} given args: {}'.
+                              format(func_name,*arg_types), node))
         else:
-            node.type_constraints = TypeInfo(return_type)
+            return_info = TypeInfo(return_type)
+        return return_info
+
+    def visit_binop(self, node):
+        t1 = node.left.type_constraints.type
+        t2 = node.right.type_constraints.type
+        op_name = op_to_dunder_binary(node.op)
```

Comment:
Why don't you put this in the same helper function you created above? This is nice because you can use the number of arguments to determine whether to call `op_to_dunder_binary` or the unary version.